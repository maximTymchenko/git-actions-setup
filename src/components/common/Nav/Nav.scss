@import '../../../styles/vars';
.nav {
    display: flex;
    align-items: flex-start;

    @media screen and (max-width: $tablet) {
        align-items: flex-end;
    }

    &--main {
        @media screen and (max-width: $mobile) {
            overflow-x: scroll;
            width: 100%;
            max-width: 600px;
            padding-left: 200px;
            position: relative;
            left: 25px;
        }
    }

    &--filter {
        @include main__padding;

        @media screen and (max-width: $desktop) {
            overflow-x: scroll;
            width: 100%;
            max-width: 800px;
        }

        @media screen and (max-width: $small__desktop) {
            // overflow-x: scroll;
            width: 100%;
            max-width: 690px;
            position: relative;
            right: 50px;
        }

        @media screen and (max-width: $tablet) {
            max-width: 590px;
            right: 60px;
        }

        @media screen and (max-width: $small__tablet) {
            max-width: 540px;
            right: 30px;
        }

        @media screen and (max-width: $large__mobile) {
            max-width: 420px;
            // right: 40px;
        }

        @media screen and (max-width: $mobile) {
            max-width: 420px;
            right: -30px;
            padding-left: 200px;
        }
    }

    &__container {
        display: flex;

        @media screen and (max-width: $mobile) {
            position: relative;
            left: -200px;
        }
    }

    &--mobile {
        @media screen and (max-width: $mobile) {
            width: 100%;
            height: 100%;
            background-color: $background__primary;
            flex-direction: column;
            position: fixed;
            right: -70px;
            top: 70px;
            border-left: 2px solid $secondary;
        }
    }

    &__link {
        cursor: pointer;
        font-family: 'OktaNeue-Bold';
        font-size: 15px;
        font-weight: bold;
        text-align: center;
        color: $secondary;
        @include transition;
        // margin-top: 10px;
        margin-right: 15px;

        @media screen and (max-width: $desktop) {
            margin-right: 0;
        }

        &:hover {
            color: $primary__light;
        }
        &:active {
            color: $text__body;
        }

        @media screen and (max-width: $tablet) {
            margin-right: 15px;
        }

        @media screen and (max-width: $mobile) {
            text-align: left;
            width: 170%;
        }
    }
}
