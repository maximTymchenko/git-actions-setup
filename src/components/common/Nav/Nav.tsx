import React, { useState, useRef, useEffect } from 'react'
import { Link } from 'gatsby'
import './Nav.scss'
import { INavLinks } from '../PrimaryNav'
import Button from '../Button'
import arrLeft from '../../../assets/img/arr-left.png'

const Nav = ({ links, path, type }) => {
  const [navLinks, setNavLinks] = useState<INavLinks>(links)

  const toggleNavClass = (index: number) => {
    setNavLinks({ ...navLinks, activeLink: navLinks.links[index] })
  }

  const toggleActiveClass = (index: number) => {
    if (navLinks.links[index] === navLinks.activeLink) {
      return 'nav__link nav__link--active'
    } else {
      return 'nav__link'
    }
  }

  //   const linkRef = useRef()

  //   const [linkState, setLinkState] = useState<boolean>(false)

  //   const getElemClass = () => {
  //     const linkClasses = linkRef.current
  //     // console.log();

  //     if (linkClasses.classList.contains('active__border--style')) {
  //       setLinkState(true)
  //       console.log('YES')
  //     }
  //   }

  //   console.log(linkRef)

  return (
    <ul className={`nav nav--${type}`}>
      <div className="nav__container">
        {/* <Button icon={arrLeft} type="apply">
                    Back
                </Button> */}
        {navLinks.links.map((item, index) => (
          <Link
            // ref={linkRef}
            data-testid="nav"
            activeClassName="active__border--style"
            to={`${item.page}`}
            className="link__margin  active__border nav__link"
            state={{ choice: path }}
            // onClick={getElemClass}
          >
            <li
              onClick={() => {
                toggleNavClass(index)
              }}
              className={toggleActiveClass(index)}
              key={index}
            >
              {item.title}
            </li>

            {/* {item.title} */}
          </Link>
        ))}
      </div>
    </ul>
  )
}

export default Nav
