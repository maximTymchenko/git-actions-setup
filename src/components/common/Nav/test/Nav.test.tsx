import React from 'react'
import Nav from '../Nav'
import { queryAllByTestId, render } from '@testing-library/react'
import renderer from 'react-test-renderer'

const menuLinks = {
  activeLink: null,
  links: [
    { id: 1, title: 'All' },
    { id: 2, title: 'Web development' },
    { id: 3, title: 'Cloud services' },
    { id: 4, title: 'Team extension' },
    { id: 5, title: 'AR & VR' },
    { id: 6, title: 'Retail & E-commerce' },
    { id: 7, title: 'Mobile development' },
  ],
}

it('renders elem correctly', () => {
  const { queryAllByTestId } = render(<Nav links={menuLinks} path="test"></Nav>)
  expect(queryAllByTestId('nav')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(<Nav links={menuLinks} path="test"></Nav>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
