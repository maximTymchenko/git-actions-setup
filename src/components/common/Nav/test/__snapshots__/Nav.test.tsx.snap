// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<ul
  className="nav nav--undefined"
>
  <div
    className="nav__container nav__container--undefined"
  >
    <a
      className="link__margin  active__border nav__link undefined"
      data-testid="nav"
      href="/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <li
        className="nav__link"
        onClick={[Function]}
      >
        All
      </li>
    </a>
    <a
      className="link__margin  active__border nav__link undefined"
      data-testid="nav"
      href="/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <li
        className="nav__link"
        onClick={[Function]}
      >
        Web development
      </li>
    </a>
    <a
      className="link__margin  active__border nav__link undefined"
      data-testid="nav"
      href="/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <li
        className="nav__link"
        onClick={[Function]}
      >
        Cloud services
      </li>
    </a>
    <a
      className="link__margin  active__border nav__link undefined"
      data-testid="nav"
      href="/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <li
        className="nav__link"
        onClick={[Function]}
      >
        Team extension
      </li>
    </a>
    <a
      className="link__margin  active__border nav__link undefined"
      data-testid="nav"
      href="/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <li
        className="nav__link"
        onClick={[Function]}
      >
        AR & VR
      </li>
    </a>
    <a
      className="link__margin  active__border nav__link undefined"
      data-testid="nav"
      href="/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <li
        className="nav__link"
        onClick={[Function]}
      >
        Retail & E-commerce
      </li>
    </a>
    <a
      className="link__margin  active__border nav__link undefined"
      data-testid="nav"
      href="/undefined"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <li
        className="nav__link"
        onClick={[Function]}
      >
        Mobile development
      </li>
    </a>
  </div>
</ul>
`;
